cpp/src/arrow/flight/server.cc

```diff
@@ -240,6 +273,90 @@ class GrpcServerAuthSender : public ServerAuthSender {
   grpc::ServerReaderWriter<pb::HandshakeResponse, pb::HandshakeRequest>* stream_;
 };
 
+class GrpcMessageWriter : public FlightMessageWriter {
```

Comment:
Is there possibly a clearer name for this class?

Reply:
I renamed this `DoExchangeMessageWriter` since that's the only place it's used.